src/backend/embedding/chunkfile_Weaviateprocess/chunkfile_uploadWeaviate.py

```diff
@@ -0,0 +1,115 @@
+import logging
+import json
+from weaviate import Client
+from sentence_transformers import SentenceTransformer
+from src.backend.weaviate.weaviate_manager import WeaviateManager
+
+# Configure logging
+logging.basicConfig(level=logging.DEBUG, format="%(asctime)s - %(levelname)s - %(message)s")
+logger = logging.getLogger(__name__)
+
+class WeaviateSearch:
+    def __init__(self):
+        try:
+            self.client = Client("http://localhost:8080")
+            logger.info("Successfully connected to Weaviate")
+
+            self.model = SentenceTransformer("all-MiniLM-L6-v2")
+            logger.info("Successfully connected to Weaviate with external embedding model.")
+
+            self.weaviate=WeaviateManager()
+
+        except Exception as e:
+            logger.error(f"Failed to initialize Weaviate connection: {e}")
+            raise e
+
+    def insert_data(self, customer_guid: str, file_path: str):
+
+        try:
+            class_names=self.weaviate.generate_weaviate_class_name(customer_guid)
+
```

Comment:
Should we check whether class name exists and if not we should add the class first before inserting data into weaviate?

Should we use the function we have already written create_schema_per_customer_guid(self, customer_guid):